src/main/java/org/apache/commons/lang3/ThreadUtils.java

```diff
@@ -0,0 +1,341 @@
+
+        final Thread[] threads = findThreadsByName(threadName);
+
+        if(threads.length == 0) {
+            return EMPTY_THREAD_ARRAY;
+        }
+
+        final List<Thread> matchedThreads = new ArrayList<Thread>();
+
+        for (int i = 0; i < threads.length; i++) {
+            final Thread thread = threads[i];
+
+            if(thread != null && ArrayUtils.contains(threadGroups, thread.getThreadGroup())) {
+                matchedThreads.add(thread);
+            }
+
+        }
+        return matchedThreads.toArray(new Thread[matchedThreads.size()]);
+    }
+
+    /**
+     * Return active thread groups with the specified group name
+     *
+     * @param threadGroupName The thread group name
+     * @return the thread groups with the specified group name or an empty array if no such thread group exists
+     * @throws IllegalArgumentException if group name is null
+     * @throws  SecurityException
+     *          if the current thread cannot access the system thread group
+     *
+     * @throws  SecurityException  if the current thread cannot modify
+     *          thread groups from this thread's thread group up to the system thread group
+     */
+    public static ThreadGroup[] findThreadGroupsByName(final String threadGroupName) {
+        if (threadGroupName == null) {
+            throw new IllegalArgumentException("The threadGroupName must not be null");
+        }
+        final ThreadGroup[] allThreadGroups = getAllThreadGroups();
+        final List<ThreadGroup> threadGroups = new ArrayList<ThreadGroup>();
+        for (int i = 0; i < allThreadGroups.length; i++) {
+            final ThreadGroup threadGroup = allThreadGroups[i];
+            if (threadGroup != null && threadGroupName.equals(threadGroup.getName())) {
+                threadGroups.add(threadGroup);
+            }
+        }
+        return threadGroups.toArray(new ThreadGroup[threadGroups.size()]);
+    }
+
+    /**
+     * Return all active thread groups including the system thread group (A thread group is active if it has been not destroyed)
+     *
+     * @return all thread groups including the system thread group
+     * @throws  SecurityException
+     *          if the current thread cannot access the system thread group
+     *
+     * @throws  SecurityException  if the current thread cannot modify
+     *          thread groups from this thread's thread group up to the system thread group
+     */
+    public static ThreadGroup[] getAllThreadGroups() {
+        final ThreadGroup systemThreadGroup = getSystemThreadGroup();
+        int estimatedThreadGroupCount = systemThreadGroup.activeGroupCount();
+        int threadGroupCount = 0;
+        ThreadGroup[] threadGroups;
+        do {
+            estimatedThreadGroupCount *= 2;
+            threadGroups = new ThreadGroup[estimatedThreadGroupCount];
+            threadGroupCount = systemThreadGroup.enumerate(threadGroups);
+        } while (threadGroupCount == estimatedThreadGroupCount);
```

Comment:
this comparison is wrong, shouldnt it be threadGroupCount > estimatedThreadGroupCount?